src/mito2/src/worker.rs

```diff
@@ -638,6 +649,26 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             let _ = sender.send(SetReadonlyResponse::NotFound);
         }
     }
+
+    async fn edit_region(&self, region_id: RegionId, edit: RegionEdit) -> Result<()> {
+        let region = self
+            .regions
+            .get_region(region_id)
+            .context(RegionNotFoundSnafu { region_id })?;
```

Comment:
We should use `writable_region()` to get a writable region. We should not modify the region when the region is readonly.